### PR TITLE
Variants: add common concept for include/require

### PR DESCRIPTION
**Pull request by tobsan**
Don't merge until [meta-pelux/54](https://github.com/Pelagicore/meta-pelux/pull/54) has been merged.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

### CODE REVIEW

File: docs/articles/baseplatform/variant-concept-in-yocto.rst [hunk 1]
```diff
@@ -36,6 +36,15 @@ your layer. The variant directories should go into
   as ``bblayers.conf.sample``, but should contain the ``local.conf`` needed for
   the current variant.
 
+If many variants are using the same directives, it is recommended to have a
+``common`` directory with these directives in files that can then be included by
```
Comment (mardy) [inline docs/articles/baseplatform/variant-concept-in-yocto.rst]: This is a bit hard to read. Maybe "...a ``common`` directory containing configuration files with these shared directives. These files can then...".
It's still a bit convoluted, but I cannot come up with anything better.

**Review by mardy — CHANGES_REQUESTED** [review]

File: docs/articles/baseplatform/variant-concept-in-yocto.rst (hunk 1 above)
Comment (tobsan) [reply docs/articles/baseplatform/variant-concept-in-yocto.rst]: Updated.

File: docs/articles/baseplatform/variant-concept-in-yocto.rst [hunk 2]
```diff
@@ -21,7 +21,7 @@ Recommended variant management
 
 The recommended way to handle variants is to keep one directory per variant in
 your layer. The variant directories should go into
-``<meta-yourlayer>/conf/<variant name>``. Each variant should supply a
+``<meta-yourlayer>/conf/variant/<variant name>``. Each variant should supply a
```
Comment (jonte) [inline docs/articles/baseplatform/variant-concept-in-yocto.rst:24]: I'd prefer the shorter path, as it was.. How set are you on this? ;)

File: docs/articles/baseplatform/variant-concept-in-yocto.rst [hunk 3]
```diff
@@ -36,6 +36,16 @@ your layer. The variant directories should go into
   as ``bblayers.conf.sample``, but should contain the ``local.conf`` needed for
   the current variant.
 
+If many variants are using the same directives, it is recommended to have a
+``common`` directory with these directives in. The files that contain those
+common directives can then be included by the more specific variants that add
+the specific settings.
+
+.. note:: BitBake searches paths that are listed in the ``BBPATH`` variable, so
+          before including any files in bblayers.conf or local.conf, one has to
```
Comment (jonte) [inline docs/articles/baseplatform/variant-concept-in-yocto.rst]: Backticks around filenames

**Review by jonte — APPROVED** [review]
Looks OK, if you want to keep the ``conf/variant/<variant name>`` convention.

File: docs/articles/baseplatform/variant-concept-in-yocto.rst (hunk 2 above)
Comment (jeena) [reply docs/articles/baseplatform/variant-concept-in-yocto.rst:24]: The shorter path doesn't reflect the structure with what was in the directory before, @tobsan had a good point there.

**Pull request merged**